packages/airnode-deployer/src/infrastructure/index.ts

```diff
@@ -460,18 +460,46 @@ async function fetchDeployments(cloudProviderType: CloudProvider['type'], deploy
       }
 
       const latestDeployment = Object.keys(stageDirectory.children).sort().reverse()[0];
+      // S
+      const latestDepolymentFileNames = Object.keys(
+        (stageDirectory.children[latestDeployment] as Directory)?.children || {}
+      );
+
+      const requiredFileNames = ['config.json', 'secrets.env', 'default.tfstate'];
+      const missingRequiredFiles = requiredFileNames.filter(
+        (requiredFileName) => !latestDepolymentFileNames.includes(requiredFileName)
+      );
+      if (missingRequiredFiles.length) {
+        logger.warn(
+          `Airnode '${airnodeAddress}' with stage '${stage}' is missing files: ${missingRequiredFiles.join(
+            ', '
+          )}. Deployer commands may fail and manual removal may be necessary.`
+        );
```

Comment:
Shouldn't we `continue` and skip this loop if this happens?

Reply:
Good point :smile: